apps/ideas/forms.py

```diff
@@ -170,3 +189,134 @@ class Meta:
+                    help_text=COLLABORATORS_EDIT_HELP,
+                    label=COLLABORATORS_EDIT_TITLE,
+                    choices=[
+                        (c, c) for c in collaborators
+                    ],
+                    initial=collaborators,
+                    widget=forms.CheckboxSelectMultiple
+                )
+
+            invites = self.invite_emails
+            if invites:
+                self.fields['invites'] = forms.MultipleChoiceField(
+                    required=False,
+                    help_text=INVITES_EDIT_HELP,
+                    label=INVITES_EDIT_TITLE,
+                    choices=[
+                        (i, i) for i in invites
+                    ],
+                    initial=invites,
+                    widget=forms.CheckboxSelectMultiple
+                )
+            self.fields.move_to_end('invites', last=False)
+            self.fields.move_to_end('collaborators', last=False)
+            self.fields.move_to_end('collaborators_emails', last=False)
+
+    @property
+    def helper(self):
+        helper = super().helper
+        helper['collaborators'].wrap(
+            crisp.layout.Field,
+            wrapper_class='checkboxes-delete',
+            css_class='checkbox-delete'
+        )
+        helper['invites'].wrap(
+            crisp.layout.Field,
+            wrapper_class='checkboxes-delete',
+            css_class='checkbox-delete'
+        )
+        return helper
+
+    @property
+    def collaborator_names(self):
+        return self.instance.collaborators.all().values_list(
+            'username', flat=True
+        )
+
+    @property
+    def invite_emails(self):
+        return self.instance.ideainvite_set.values_list('email', flat=True)
+
+    def clean(self):
+        super().clean()
+
+        addresses = self.cleaned_data.get('collaborators_emails', [])
+        invites = self.cleaned_data.get('invites', [])
+        collaborators = self.cleaned_data.get('collaborators', [])
+
+        duplicate_errors = []
+        for (name, address) in addresses:
+            if address in invites:
+                error = ValidationError({
+                   'collaborators_emails': _(
+                       'You already invited {email}'
+                   ).format(email=address)
+                })
+                duplicate_errors.append(error)
+        if duplicate_errors:
+            raise ValidationError(duplicate_errors)
+
+        collaborator_count = sum([
+            len(addresses),
+            len(invites),
+            len(collaborators),
+        ])
+
+        if collaborator_count > 5:
+            raise ValidationError(_('Maximum 5 collaborators allowed'))
+
+    def save(self, commit=True):
+        """
+        Deletes invites and collaborators and adds new invites of instance.
+
+        There is a little hack here, it uses the idea creator and not the
+        current user as creator for the invites. There for no user needs to
```

Comment:
I think that's fine as the whole idea is also created by that creator and the collaborators are also only a model field after all.